File: backend/utils/PermissionChecker.go
package utils

import "github.com/lxndrrud/blog-project/models"

type IPermissionChecker interface {
	CanCreatePosts(permissions []models.Permission) bool
}

func NewPermissionChecker() IPermissionChecker {
	return &permissionChecker{}
}

type permissionChecker struct {
}

func (c *permissionChecker) CanCreatePosts(permissions []models.Permission) bool {
	for _, p := range permissions {
		if p.Code == "СОЗД_ПОСТЫ" {
			return true
		}
	}
	return false
}

File: backend/src/utils/PermissionChecker.ts
import { TPermission } from "../types/Permission.type";

export interface IPermissionChecker {
    canCreatePosts(permissionList: TPermission[]): boolean
    canModeratePosts(permissionList: TPermission[]): boolean
}

export class PermissionChecker implements IPermissionChecker {
    public canCreatePosts(permissionList: TPermission[]) {
        for (let permission of permissionList) {
            if (permission.code === "СОЗД_ПОСТЫ") return true
        }
        return false
    }

    public canModeratePosts(permissionList: TPermission[]) {
        for (let permission of permissionList) {
            if (permission.code === "МОДЕР_ПОСТЫ") return true
        }
        return false
    }
}
